### PR TITLE
Common address component

### PR DESCRIPTION
**Pull request by anuroop15**
This PR adds Common Address Component to react-wood-duck library.

Please run npm install before prepublish, for installing 'react-autosuggest' component.
reference: https://github.com/moroshko/react-autosuggest

### CODE REVIEW

**Comment by ratneshraval** [comment]
I recommend sitting down with Ed or Pradeep to resolve this conflicts.

File: src/test/CommonAddressComponent_tests.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      stateTypes={stateTypes.items}
+      addressFields={addressFields}
+      onChange={onChangeSpy}
+      renderSuggestion={onChangeSpy}
+      onSuggestionsFetchRequested={onChangeSpy}/>)
+  })
+  it('Load Address Fields', () => {
+    expect(commonAddressComp.length).toEqual(1 )
+  })
+  describe('Streed Address', () => {
```
Comment (edgibbs) [inline src/test/CommonAddressComponent_tests.js]: Don't understand the describe block use here versus all the rest of the 'it' blocks for consistency.

File: src/test/CommonAddressComponent_tests.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    let cityField = commonAddressComp.find('#Residentialcity').at(0)
+    cityField.simulate('change', {target: {value: 'Sacramento'}})
+    expect(onChangeSpy).toHaveBeenCalledWith('city', 'Sacramento')
+  })
+  it('verify mailing Address State change', () => {
+    let stateDropDownField = commonAddressComp.find('#Residentialstate_type').at(0)
+    stateDropDownField.simulate('change', {target: {selectedOptions: [{value: 1, text: 'Alabama'}]}})
+    expect(onChangeSpy).toHaveBeenCalledWith('state', { id: 1, value: 'Alabama' })
+  })
+  describe('with Test Utils', () => {
```
Comment (edgibbs) [inline src/test/CommonAddressComponent_tests.js]: This is describing the testing tool specific and it should probably be 'with AutoSuggest'.

File: src/test/CommonAddressComponent_tests.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    let renderedComp, compRendered, resultTag
+    beforeEach(() => {
+      renderedComp = TestUtils.createRenderer()
+      compRendered = renderedComp.render(<CommonAddressComponent {...props}/>)
```
Comment (edgibbs) [inline src/test/CommonAddressComponent_tests.js]: compRendered is never used

File: src/test/CommonAddressComponent_tests.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  })
+  it('verify mailing Address State change', () => {
+    let stateDropDownField = commonAddressComp.find('#Residentialstate_type').at(0)
+    stateDropDownField.simulate('change', {target: {selectedOptions: [{value: 1, text: 'Alabama'}]}})
+    expect(onChangeSpy).toHaveBeenCalledWith('state', { id: 1, value: 'Alabama' })
+  })
+  describe('with Test Utils', () => {
+    let renderedComp, compRendered, resultTag
+    beforeEach(() => {
+      renderedComp = TestUtils.createRenderer()
```
Comment (edgibbs) [inline src/test/CommonAddressComponent_tests.js]: Isn't this a renderer, makes a much better name.

File: src/test/TextArea_test.js [hunk 5]
```diff
@@ -13,7 +13,12 @@ describe('TextArea', function() {
     expect(resultTag.type).toBe('div');
   });
 
-  it('verify the className', function() {
-    expect(resultTag.props.className).toBe('form-group');
-  });
-});
+    it('renders the tag', function() {
+        expect(resultTag.type).toBe('div')
```
Comment (edgibbs) [inline src/test/TextArea_test.js]: Duplicate test.  Is this a cut and paste error or a merge error?

Comment (ratneshraval) [reply src/test/TextArea_test.js]: Could be a merge error that I did. Anuroop, can you remove this with your changes.

File: src/CommonAddressComponent.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    this.getSuggestionValue = this.getSuggestionValue.bind(this);
+    this.renderSuggestion = this.renderSuggestion.bind(this);
+    this.onStreetAddressChange = this.onStreetAddressChange.bind(this);
+  }
+
+  getSuggestionValue(suggestion) {
+    return suggestion.street_address;
+  }
+  onSuggestionsClearRequested() {
+    // this.props.onSelection(suggestion)
```
Comment (edgibbs) [inline src/CommonAddressComponent.js]: Why is this commented out?  Does it need to clear out the suggestion state?

Comment (anuroop15) [reply src/CommonAddressComponent.js]: Right!! But we are not doing any clearing out condition as of design. And as  onSuggestionsClearRequested is mandatory for AutoSuggest plugin, I have placed it with commented lines.

Comment (anuroop15) [reply src/CommonAddressComponent.js]: I can change the wording of commented lines if needed.

Comment (edgibbs) [reply src/CommonAddressComponent.js]: How about just leaving the function with an empty body?

**Comment by ratneshraval** [comment]
@pierreOSI Can you approve this as well?

**Pull request merged**